lib/src/cache.rs

```diff
@@ -159,6 +164,24 @@ impl CacheEntry {
     pub fn take_go_get(&mut self) -> Option<Obligation> {
         self.go_get.take()
     }
+
+    /// Insert the provided body into the cache.
+    pub fn insert(&mut self, options: WriteOptions, body: Body) -> Result<(), crate::Error> {
+        let go_get = self.take_go_get().ok_or(Error::NotRevalidatable)?;
+        go_get.insert(options, body);
+        Ok(())
+    }
+
+    /// Freshen the existing body according to the provided options.
```

Comment:
Not a blocker, but the wording here could be a bit misleading. Maybe: `Freshen the existing cache item according to the provided options, without changing the body`